Marlin/src/inc/SanityCheck.h

```diff
@@ -1672,6 +1672,11 @@ static_assert(NUM_SERVOS <= NUM_SERVO_PLUGS, "NUM_SERVOS (or some servo index) i
   #endif
 #endif
 
+#define WRONG_CURRENT_HOME(N) (N##_CURRENT_HOME > N##_CURRENT) ||
+  #if MAIN_AXIS_MAP(WRONG_CURRENT_HOME) MAP(WRONG_CURRENT_HOME, X2, Y2, Z2, Z3, Z4) 0
+    #error "One of CURRENT_HOME needs to be inferior or equal to its CURRENT"
+  #endif
```

Comment:
While my personal builds usually halve run current for the homing current, why limit this? Maybe someone needs a larger homing current.

Reply:
After evaluation, there was no practical use-case that would require higher homing current than running current. It was concluded that this would guard user against setting incorrect homing currents from either a typo or re-configuring running current and forgetting about custom homing currents. It is a fail-safe approach to prevent mechanical damage, should something be set wrong

Homing current has its best use-case for sensorless homing, where low currents provide better and cleaner homing

I suggest that a warning about setting very high running currents (>2A) would also be a good guiding step for inexperienced users, either as SanityCheck or, preferably, in the `Configuration_adv.h` file. Especially since most stepper drivers can handle a maximum of 1.6A RMS anyway

Reply:
> After evaluation, there was no practical use-case that would require higher homing current than running current.

I can't keep stressing this enough: **_You have to think of more than just your use case._**

This may be required for some configs. I don't think we should just arbitrarily limit this until there's an issue.

> I suggest that a warning about setting very high running currents (>2A) would also be a good guiding step for inexperienced users, either as SanityCheck or, preferably, in the Configuration_adv.h file. Especially since most stepper drivers can handle a maximum of 1.6A RMS anyway

Instead of setting generic sanity checks like this, it'd be better to find the max values for various drivers and set them correctly where possible. There are some high-powered drivers out there like TMC5160 (and soon to be supported TMC2240) that can run at > 2 amps. Another example: BigTreeTech has several 5160 variants that can run from 4-10amps.

Reply:
And you need to start thinking about the actual user base of Marlin that is mostly inexperienced and needs guiding. Not everybody using Marlin is proficient at printer firmware. An experienced user will know how to bypass the warning anyway.

Sanity checks are here to keep the user from incompatible choices or dangerous situations. They are not here just to please the CPP compiler. Setting higher currents for homing by mistake is a dangerous situation where the printer could get damaged. It is in the exact same nature as all other checks that prevents movement in "impossible areas" or combination of settings that would be deemed unsafe

Reply:
> And you need to start thinking about the actual user base of Marlin that is mostly inexperienced and needs guiding.

Clearly, you have no idea what I primarily focus on around here.

I'm going to bow out of this PR review. Best of luck!

Reply:
I concur with @sargonphin that we should guide users to the best settings, but also that where there might be some counterintuitive use case, we should make the `#error` into a `#warning` or make the `#error` skippable like we do with some others.